encodings/byte_bool/src/compute/mod.rs

```diff
@@ -0,0 +1,265 @@
+use std::ops::{BitAnd, BitOr, BitXor, Not};
+use std::sync::Arc;
+
+use arrow_buffer::BooleanBuffer;
+use num_traits::AsPrimitive;
+use vortex::validity::Validity;
+use vortex::ToArrayData;
+use vortex::{
+    compute::{
+        compare::CompareFn, slice::SliceFn, take::TakeFn, unary::fill_forward::FillForwardFn,
+        unary::scalar_at::ScalarAtFn, ArrayCompute,
+    },
+    encoding::ArrayEncodingRef,
+    stats::StatsSet,
+    validity::ArrayValidity,
+    ArrayDType, ArrayData, ArrayTrait, IntoArray,
+};
+use vortex::{Array, IntoCanonical};
+use vortex_dtype::{match_each_integer_ptype, Nullability};
+use vortex_error::{vortex_bail, VortexResult};
+use vortex_expr::Operator;
+use vortex_scalar::{Scalar, ScalarValue};
+
+use super::{ByteBoolArray, ByteBoolMetadata};
+
+impl ArrayCompute for ByteBoolArray {
+    fn compare(&self) -> Option<&dyn CompareFn> {
+        Some(self)
+    }
+
+    fn fill_forward(&self) -> Option<&dyn FillForwardFn> {
+        None
+    }
+
+    fn scalar_at(&self) -> Option<&dyn ScalarAtFn> {
+        Some(self)
+    }
+
+    fn slice(&self) -> Option<&dyn SliceFn> {
+        Some(self)
+    }
+
+    fn take(&self) -> Option<&dyn TakeFn> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn for ByteBoolArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        if index >= self.len() {
+            vortex_bail!(OutOfBounds: index, 0, self.len());
+        }
+
+        let scalar = match self.is_valid(index).then(|| self.buffer()[index] == 1) {
+            Some(b) => Scalar::new(self.dtype().clone(), ScalarValue::Bool(b)),
+            None => Scalar::null(self.dtype().clone()),
+        };
+
+        Ok(scalar)
+    }
+}
+
+impl SliceFn for ByteBoolArray {
+    fn slice(&self, start: usize, stop: usize) -> VortexResult<Array> {
+        let length = stop - start;
+
+        let validity = self.validity().slice(start, stop)?;
+
+        let slice_metadata = Arc::new(ByteBoolMetadata {
+            validity: validity.to_metadata(length)?,
+        });
+
+        ArrayData::try_new(
```

Comment:
You generally want to go through the typed constructor where possible. So you want a 
```try_new(buffer: Buffer, validity: Validity) -> Self``` and then rename existing `try_with_validity` to `from_vec`?

Reply:
That makes sense, introduced `try_new` and renamed to `try_from_vec`